### PR TITLE
nixos/minio: gracefully handle root credentials file

### PR DESCRIPTION
**Pull request by jsoo1**
Otherwise the `minio.service` service will fail either:                                                   

* with a message that the EnvironmentFile does not exist                                                  
* or silently with potentially stale credentials

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bachp — APPROVED** [review]
Change looks resonable

File: nixos/modules/services/web-servers/minio.nix [hunk 1]
```diff
@@ -96,29 +96,59 @@ in
...
-        MINIO_REGION = "${cfg.region}";
-        MINIO_BROWSER = "${if cfg.browser then "on" else "off"}";
-      };
-    };
+    }
+
+      (lib.mkIf (cfg.rootCredentialsFile != null) {
+        services.minio.unitConfig.ConditionPathExists = cfg.rootCredentialsFile;
+
+        paths.minio-root-credentials = {
```
Comment (bachp) [inline nixos/modules/services/web-servers/minio.nix:129]: Can we add a comment why this service is needed.

**Review by bachp — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-servers/minio.nix (hunk 1 above)
Comment (jsoo1) [reply nixos/modules/services/web-servers/minio.nix:132]: Well. I haven't found anything that would do the trick. Added the commentary. Thank you!

Comment (bachp) [reply nixos/modules/services/web-servers/minio.nix:132]: @jsoo1 I also didn't find a better way. It's uncommen enough to warant a comment.

**Review by bachp — APPROVED** [review]

**Pull request merged**